hw02-generic/src/main/java/ru/otus/CustomerService.java

```diff
@@ -0,0 +1,23 @@
+package ru.otus;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    private final TreeMap<Customer, String> customers = new TreeMap<>();
```

Comment:
```suggestion
    private final NavigableMap<Customer, String> customers = new TreeMap<>();
```
Тип лучше объявлять минимально необходимым интерфейсом.